File: src_local/agents/Kotlin/scope.py
from src_local.agents.base import BaseActivePersona
import logging
import time
from pathlib import Path

logger = logging.getLogger(__name__)

class ScopePersona(BaseActivePersona):
    """
    Core: PhD in Product Engineering & Technical Roadmap 🔭
    Monitorado por Metric: Injeção de Telemetria de Auditoria.
    """
    
    def __init__(self, project_root):
        super().__init__(project_root)
        self.name, self.emoji, self.role, self.stack = "Scope", "🔭", "PhD Product Architect (Kotlin)", "Kotlin"

    def perform_audit(self) -> list:
        """Auditoria com telemetria de alinhamento técnico integrada."""
        start_time = time.time()
        logger.info(f"[{self.name}] Analisando Backlog Técnico...")
        
        # Sintaxe linear
        rules = [
            {'regex': r"//\s*TODO", 'issue': 'Débito Técnico: Marcador detectado.', 'severity': 'medium'},
            {'regex': r"debuggable\s+true", 'issue': 'Risco Crítico: App debuggable em produção.', 'severity': 'high'}
        ]
        
        results = self.find_patterns(('.kt', '.gradle', '.kts'), rules)
        self._log_performance(start_time, len(results))
        return results

    def _reason_about_objective(self, objective, file, content):
        # O Scope agora delega a auditoria de escopo para o AuditEngine via perform_audit
        return None

    def get_system_prompt(self):
        return f"Você é o Dr. {self.name}, guardião da entrega JVM."

File: src_local/agents/Kotlin/Audit/scope.ts
import { BaseActivePersona } from "../../base.ts";
import type { AuditRule, StrategicFinding } from "../../base.ts";
import type { ProjectContext } from "../../../core/types.ts";

/**
 * 🎯 Dr. Scope — PhD in Kotlin Project Management & Technical Debt
 */
export class ScopePersona extends BaseActivePersona {
    constructor(projectRoot: string | undefined = undefined) {
        super(projectRoot);
        this.name = "Scope";
        this.emoji = "🎯";
        this.role = "PhD Project Strategist";
        this.phd_identity = "Kotlin Project Management & Technical Debt";
        this.stack = "Kotlin";
    }

    override async execute(context: any): Promise<any> {
        this.setContext(context);
        const findings = await this.performAudit();
        return findings;
    }

    override getAuditRules(): { extensions: string[]; rules: AuditRule[] } {
        return {
            extensions: [".kt"],
            rules: [
                { regex: /\/\/\s*TODO[:\s]/i, issue: 'Dívida Técnica: TODO pendente no código Kotlin.', severity: 'medium' }
            ]
        };
    }

    public audit(): any[] { return []; }
    public Branding(): string { return `${this.emoji} ${this.name}`; }
    public Analysis(): string { return "Project Scope Analysis Complete"; }
    public test(): boolean {
        this.Branding();
        this.Analysis();
        this.audit();
        return true;
    }

    override reasonAboutObjective(objective: string, file: string, content: string | Promise<string | null>): StrategicFinding | string | null {
        return {
            file, severity: "INFO",
            issue: `PhD Scope (Kotlin): Analisando escopo para ${objective}.`,
            context: "analyzing project debt"
        };
    }

    override selfDiagnostic(): any {
        return { status: "Soberano", score: 100, issues: [], branding: this.Branding() };
    }

    override getSystemPrompt(): string {
        return `Você é o Dr. ${this.name}, PhD em dívida técnica Kotlin. Status: ${this.Analysis()}`;
    }
}
